tools/topology/topology2/platform/intel/dmic-aec.conf

```diff
@@ -0,0 +1,363 @@
+				name 'AEC Capture Volume 1'
+			}
+		}
+		Object.Widget.copier."2" {
+			Object.Base.audio_format.1 {
+				in_bit_depth		32
+				in_valid_bit_depth	32
+				out_bit_depth		32
+				out_valid_bit_depth	32
+				dma_buffer_size "$[$ibs * 2]"
+			}
+			Object.Base.audio_format.2 {
+				in_channels		4
+				in_bit_depth		32
+				in_valid_bit_depth	32
+				out_channels		4
+				out_bit_depth		32
+				out_valid_bit_depth	32
+				dma_buffer_size "$[$ibs * 2]"
+				in_ch_cfg	$CHANNEL_CONFIG_3_POINT_1
+				in_ch_map	$CHANNEL_MAP_3_POINT_1
+				out_ch_cfg	$CHANNEL_CONFIG_3_POINT_1
+				out_ch_map	$CHANNEL_MAP_3_POINT_1
+			}
+		}
+	}
+
+	dai-copier-gain-module-copier-capture.2 {
+		index		$DMIC0_DAI_PIPELINE_ID
+		Object.Widget.pipeline.1 {
+			stream_name $DMIC0_NAME
+		}
+
+		Object.Widget.copier.1 {
+			dai_index	0
+			type		"dai_out"
+			dai_type	"DMIC"
+			copier_type	"DMIC"
+			type		dai_out
+			stream_name	$DMIC0_NAME
+			node_type $DMIC_LINK_INPUT_CLASS
+			Object.Base.audio_format.1 {
+				in_bit_depth		32
+				in_valid_bit_depth	32
+				out_bit_depth		32
+				out_valid_bit_depth	32
+				dma_buffer_size "$[$ibs * 2]"
+			}
+			Object.Base.audio_format.2 {
+				in_channels		4
+				in_bit_depth		32
+				in_valid_bit_depth	32
+				out_channels		4
+				out_bit_depth		32
+				out_valid_bit_depth	32
+				dma_buffer_size "$[$ibs * 2]"
+				in_ch_cfg	$CHANNEL_CONFIG_3_POINT_1
+				in_ch_map	$CHANNEL_MAP_3_POINT_1
+				out_ch_cfg	$CHANNEL_CONFIG_3_POINT_1
+				out_ch_map	$CHANNEL_MAP_3_POINT_1
+			}
+		}
+
+		Object.Widget.copier."2" {
+			stream_name $DMIC0_NAME
+			Object.Base.audio_format.1 {
+				in_bit_depth		32
+				in_valid_bit_depth	32
+				out_bit_depth		32
+				out_valid_bit_depth	32
+				dma_buffer_size "$[$ibs * 2]"
+			}
+			Object.Base.audio_format.2 {
+				in_channels		4
+				in_bit_depth		32
+				in_valid_bit_depth	32
+				out_channels		4
+				out_bit_depth		32
+				out_valid_bit_depth	32
+				dma_buffer_size "$[$ibs * 2]"
+				in_ch_cfg	$CHANNEL_CONFIG_3_POINT_1
+				in_ch_map	$CHANNEL_MAP_3_POINT_1
+				out_ch_cfg	$CHANNEL_CONFIG_3_POINT_1
+				out_ch_map	$CHANNEL_MAP_3_POINT_1
+			}
+		}
+
+		Object.Widget.gain.1 {
+			num_audio_formats 2
+
+			Object.Base.audio_format.1 {
+				in_bit_depth		32
+				in_valid_bit_depth	32
+				out_bit_depth		32
+				out_valid_bit_depth	32
+				dma_buffer_size "$[$ibs * 2]"
+			}
+			Object.Base.audio_format.2 {
+				in_channels		4
+				in_bit_depth		32
+				in_valid_bit_depth	32
+				out_channels		4
+				out_bit_depth		32
+				out_valid_bit_depth	32
+				dma_buffer_size "$[$ibs * 2]"
+				in_ch_cfg	$CHANNEL_CONFIG_3_POINT_1
+				in_ch_map	$CHANNEL_MAP_3_POINT_1
+				out_ch_cfg	$CHANNEL_CONFIG_3_POINT_1
+				out_ch_map	$CHANNEL_MAP_3_POINT_1
+			}
+			Object.Control.mixer.1 {
+				name 'Main Capture Volume'
+			}
+		}
+	}
+}
+
+Object.Base {
+	route.100 {
+		source	copier.module.14.2
+		sink	copier.module.13.2
+	}
+
+	route.101 {
+		source	copier.module.14.2
+		sink	copier.module.18.2
```

Comment:
you need to explain why we have those copier to copier routes. In your own words

this is because of the restriction with the ref FW that the gain module can only be connected to modules that are in the same pipeline

Reply:
sure, thanks